src/core/server/saved_objects/service/lib/extensions/security.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EcsEventOutcome } from '@kbn/logging';
+import { SavedObject } from '../../../types';
+
+export interface CheckAuthorizationParams<A extends string> {
+  types: Set<string>;
+  spaces: Set<string>;
+  actions: A[];
+  options?: {
+    allowGlobalResource?: boolean;
+  };
+}
```

Comment:
Sorry for the lack of TSdocs here, I was trying to throw this PR together as quickly as possible 😅 

Now, the authorization check queries Elasticsearch for all permutations of specified types/spaces/actions.

When Kibana registers application privileges with Elasticsearch, each space ID corresponds to a specific _resource_. Instead of checking privileges for individual resources, you can choose to check privileges for "*", which is the _global resource_. When you create a role that grants access to "All spaces", under the hood it's granting the specified feature privileges for the global resource.

It's important to know that you can't check privileges for individual spaces _and_ the global resource at the same time. Checking privileges for the global resource supersedes any individual space.

It's also important to know that if a saved object exists in "*" (all spaces), a user _doesn't_ need to have access to the global resource to be authorized to access that saved object, they only need to be authorized to access one space.

For most SOR functions, we don't want to risk accidentally checking privileges for the global resource. For example, when you `get` an object in the Default space and it exists in `namespaces: ['*']`, the SOR will check privileges for `spaces: ['default', '*']`. By default, the security extension will omit '*' from that privilege check, because we are really only interested in knowing if the user has access to the Default space.

If the `allowGlobalResource: true` option is included, the security extension will not omit '*'. This is only used when the user is attempting to (1) share an object to all spaces, or (2) create an object that exists in all spaces -- in both of these cases, we actually want to ensure the user has access to the global resource.